spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractBaseMessageHandler.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.handler;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.springframework.core.Ordered;
+import org.springframework.integration.IntegrationPattern;
+import org.springframework.integration.IntegrationPatternType;
+import org.springframework.integration.context.IntegrationObjectSupport;
+import org.springframework.integration.context.Orderable;
+import org.springframework.integration.support.management.AbstractMessageHandlerMetrics;
+import org.springframework.integration.support.management.ConfigurableMetricsAware;
+import org.springframework.integration.support.management.DefaultMessageHandlerMetrics;
+import org.springframework.integration.support.management.IntegrationManagedResource;
+import org.springframework.integration.support.management.TrackableComponent;
+import org.springframework.integration.support.management.metrics.MeterFacade;
+import org.springframework.integration.support.management.metrics.MetricsCaptor;
+import org.springframework.integration.support.management.metrics.TimerFacade;
+import org.springframework.util.Assert;
+
+/**
+ * Base class for Message handling components that provides basic validation and error
+ * handling capabilities. Asserts that the incoming Message is not null and that it does
+ * not contain a null payload. Converts checked exceptions into runtime
+ * {@link org.springframework.messaging.MessagingException}s.
+ *
+ * @author Mark Fisher
+ * @author Oleg Zhurakousky
+ * @author Gary Russell
+ * @author Artem Bilan
+ * @author Amit Sadafule
+ * @author David Turanski
+ *
+ */
+@SuppressWarnings("deprecation")
+@IntegrationManagedResource
+public abstract class AbstractBaseMessageHandler extends IntegrationObjectSupport
```

Comment:
Well, looks like this one is not a `MessageHandler` at all.
Can we call it then something like `MessageHandlerBase` ?
I kind don't like `AbstractBase` combination and it is a bit awkward when the name of the class doesn't reflect it reality. I mean since it doesn't implement `MH`, it cannot be `AbstractMH`

Reply:
Often we name things like this `MessageHandlerSupport`

Reply:
Oh! That's even better, but I hope you agree with me that it's not an `MH` 😄 

Thanks